client/packages/system/src/Encounter/DetailView/StatusChangeButton.tsx

```diff
@@ -21,9 +21,18 @@ export const StatusChangeButton = ({
   const onStatusClick = () => {
     if (!selectedOption.value || selectedOption.value === currentStatus) return;
     onSave(selectedOption.value);
-    setSelectedOption(selected => ({ ...selected, isDisabled: true }));
   };
 
+  useEffect(() => {
+    setSelectedOption(selected =>
```

Comment:
updated to reset the selected option state whenever the status changes, rather than on button click, because status can also be updated from other places